Training/Easy/horse_racing.rb

```diff
@@ -1,18 +1,18 @@
 # Solution for https://www.codingame.com/ide/puzzle/horse-racing-duals
 class HorseRacing
+  attr_accessor :horses_count
+
+  def initialize
+    @horses_count = gets.to_i
+  end
+
   def find_difference
-    strenghts = init_strenghts
-    strenghts.each_cons(2).map { |a, b| (b - a).abs }.min
+    strengths = init_strengths
+    strengths.each_cons(2).map { |a, b| (b - a).abs }.min
   end
 
-  def init_strenghts
-    n = gets.to_i
-    strenghts_array = []
-    n.times do
-      pi = gets.to_i
-      strenghts_array << pi
-    end
-    strenghts_array.sort
+  def init_strengths
+    horses_count.times.each_with_object([]) { |_, result| result << gets.to_i }.sort
   end
 
   def start
```

Comment:
Instead of the `start` method altogether, just use your `find_difference` function:
```
obj = HorceRacing.new
answer = obj.start
puts answer
```
or even
```
answer = HorceRacing.new.start
puts answer
```